### DISCUSSION TITLE
Is there any way to force a device to use a specific router?

### DISCUSSION THREAD

**Discussion by memesalot** [Koenkk/zigbee2mqtt#10432] [Devices]
Is there any way to force a device to use a specific router?

Using the "Permit join (device)" originally worked for me, but after 12/24 hours, the devices roam to "less optimal" routes.

I have 6 bulbs in my "mancave" (concrete walls) and a Tubez router in there. I joined all the bulbs/sensors to the router, everything is great. After 12-24 hours, all but 1 bulb have migrated off the router, to random outlets/bulbs in different rooms around the house. (LQI went from ~120 on each bulb, to 6-10) When using my automations (remote/motion etc), there is a huge delay on the bulbs that have a super low lq.

If this feature does not exist, it would be nice to support forcing a device to stay on one specific router, especially when it makes no sense to roam to a device very far away with a low link quality.

**Comment by MrBamse** [comment]
Wondering the same thing, did you ever figure it out?

**Comment by DubhAd** [comment]
Sadly that's up to the devices, they pick the route and Z2M can't do anything about it.
